src/corehost/cli/ijwhost/ijwhost.cpp

```diff
@@ -0,0 +1,152 @@
+// IJW entry points are defined without the __declspec(dllexport) attribute.
+// The issue here is that the MSVC compiler links to the exact name _CorDllMain instead of their stdcall-managled names.
+// So we need to export the exact name, which __declspec(dllexport) doesn't do. The solution here is to the use a .def file on Windows.
+#define IJW_API extern "C"
+#else
+#define IJW_API SHARED_API
+#endif // _WIN32
+
+pal::hresult_t get_load_in_memory_assembly_delegate(pal::dll_t handle, load_in_memory_assembly_fn* delegate)
+{
+    pal::dll_t fxr;
+
+    pal::string_t host_path;
+    if (!pal::get_own_module_path(&host_path) || !pal::realpath(&host_path))
+    {
+        trace::error(_X("Failed to resolve full path of the current host module [%s]"), host_path.c_str());
+        return StatusCode::CoreHostCurHostFindFailure;
+    }
+
+    pal::string_t dotnet_root;
+    pal::string_t fxr_path;
+    if (!resolve_fxr_path(get_directory(host_path), &dotnet_root, &fxr_path))
+    {
+        return StatusCode::CoreHostLibMissingFailure;
+    }
+
+    // Load library
+    if (!pal::load_library(&fxr_path, &fxr))
+    {
+        trace::error(_X("The library %s was found, but loading it from %s failed"), LIBFXR_NAME, fxr_path.c_str());
+        trace::error(_X("  - Installing .NET Core prerequisites might help resolve this problem."));
+        trace::error(_X("     %s"), DOTNET_CORE_INSTALL_PREREQUISITES_URL);
+        return StatusCode::CoreHostLibLoadFailure;
+    }
+
+    // Leak fxr
+
+    auto get_delegate_from_hostfxr = (hostfxr_get_delegate_fn)pal::get_symbol(fxr, "hostfxr_get_runtime_delegate");
+    if (get_delegate_from_hostfxr == nullptr)
+        return StatusCode::CoreHostEntryPointFailure;
+
+    pal::string_t app_path;
+
+    if (!pal::get_module_path(handle, &app_path))
+    {
+        trace::error(_X("Failed to resolve full path of the current mixed-mode module [%s]"), host_path.c_str());
+        return StatusCode::LibHostCurExeFindFailure;
+    }
+    
+    return get_delegate_from_hostfxr(host_path.c_str(), dotnet_root.c_str(), app_path.c_str(), hostfxr_delegate_type::load_in_memory_assembly, (void**)delegate);
+}
+
+IJW_API BOOL STDMETHODCALLTYPE _CorDllMain(HINSTANCE hInst,
+    DWORD  dwReason,
+    LPVOID lpReserved
+)
+{
+    BOOL res = TRUE;
+
+    PEDecoder pe(hInst);
+
+    // In the following code, want to make sure that we do our own initialization before
+    // we call into managed or unmanaged initialization code, and that we perform
+    // uninitialization after we call into managed or unmanaged uninitialization code.
+    // Thus, we do DLL_PROCESS_ATTACH work first, and DLL_PROCESS_DETACH work last.
+    if (dwReason == DLL_PROCESS_ATTACH)
+    {
+        // If this is not a .NET module (has a CorHeader), shouldn't be calling _CorDllMain
+        if (!pe.HasCorHeader())
+        {
+            return FALSE;
+        }
+
+        if (pe.IsILOnly() || (!pe.HasManagedEntryPoint() && !pe.HasNativeEntryPoint()))
```

Comment:
I am not sure how valuable these `IsILOnly` checks are. The IL-only check pulls in definitions of structures like R2R header that this should not really depend on.

If we get here for IL-only image, it means something is really wrong.

I would just check for the entrypoint here, and remove the IsILOnly checks from here and other places.

Reply:
I'll go remove them.